app/src/test/shared/actions/AuthActionsTests.ts

```diff
@@ -1,49 +1,126 @@
-import { expect } from 'chai';
+import { expect } from "chai";
+import * as jwt from "jsonwebtoken";
+
 import { Sandbox } from "../../Sandbox";
-const jwt = require("jsonwebtoken");
+import { authActions } from "../../../main/shared/actions/authActions";
+import { AuthService } from "../../../main/shared/services/AuthService";
+import { mainStore as contribMainStore } from "../../../main/contrib/stores/MainStore";
+import { AuthTypeKeys } from "../../../main/shared/actionTypes/AuthTypes";
+import { createMockStore } from "../../mocks/mockStore";
+import { NotificationState, notificationStore } from "../../../main/shared/stores/NotificationStore";
 
-import { authActions, LogInProperties } from "../../../main/shared/actions/AuthActions";
+import {localStorageHandler} from "../../../main/shared/services/localStorageHandler";
 
-function getPayload(tokenProperties: any): LogInProperties {
-    const token = jwt.sign(tokenProperties, "secret");
+describe("Modelling groups actions tests", () => {
     const sandbox = new Sandbox();
-    try {
-        const spy = sandbox.dispatchSpy();
-        authActions.logIn(token, false);
-        return spy.args[0][1] as LogInProperties;
-    } finally {
+    let store: any = null;
+
+    const mockUsertokenData = {
+        sub: "test.user",
+        permissions: "*/can-login,*/countries.read,*/demographics.read,*…les.write,modelling-group:test-group/users.create",
+        roles: "*/user,modelling-group:IC-Garske/member,*/user-man…/uploader,modelling-group:test-group/user-manager",
+        iss: "vaccineimpact.org",
+        exp: Math.round(Date.now() / 1000) + 1000
+    };
+
+    const mockUsertokenDataNotActive = Object.assign({}, mockUsertokenData, {
+        permissions: "*/countries.read,*/demographics.read,*…les.write,modelling-group:test-group/users.create"
+    });
+
+    const mockUsertokenNotModeller = Object.assign({}, mockUsertokenData, {
+        roles: ""
+    });
+
+    beforeEach(() => {
+        store = createMockStore();
+    });
+
+    afterEach(() => {
         sandbox.restore();
-    }
-}
+    });
 
-describe("AuthActions", () => {
-    it("retrieves username from token", () => {
-        const payload = getPayload({ sub: "username", permissions: "", roles: "" });
-        expect(payload.username).to.equal("username");
```

Comment:
Are there tests somewhere that show tokens get properly decoded?

Reply:
montagu-webapps/app/src/test/shared/modules/JwtTokenTests.ts